tgui/packages/tgui/components/NumberInput.tsx

```diff
@@ -0,0 +1,337 @@
+import { KEY } from 'common/keys';
+import { clamp } from 'common/math';
+import { BooleanLike, classes } from 'common/react';
+import {
+  Component,
+  createRef,
+  FocusEventHandler,
+  KeyboardEventHandler,
+  MouseEventHandler,
+  RefObject,
+} from 'react';
+
+import { AnimatedNumber } from './AnimatedNumber';
+import { Box } from './Box';
+
+const DEFAULT_UPDATE_RATE = 400;
+
+type Props = Required<{
+  value: number | string;
+  minValue: number;
+  maxValue: number;
+}> &
```

Comment:
```suggestion
type Props = {
  value: number | string;
  minValue: number;
  maxValue: number;
} &
```
required is default behavior - you'd use this utility to make all the properties of an existing type required, but has no effect here

Reply:
Actually I need this because https://github.com/tgstation/tgstation/pull/81913#discussion_r1518801437

Typescript be bugging :(